kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -469,12 +469,19 @@ public void onChanged(Change<? extends DescrName> change) {
         StringBinding dateStrProp = Bindings.createStringBinding(
                 () -> {
                     if (patternProperty.get() != null) {
-                        return LocalDate.ofInstant(
-                                getViewProperties().calculator()
-                                        .getFullyQualifiedDescription(patternProperty.get().nid())
-                                        .get().instant(),
-                                ZoneId.systemDefault()
-                        ).format(DateTimeFormatter.ofPattern("MMM dd, yyyy"));
+                        long rawTime = getViewProperties().calculator()
+                                .getFullyQualifiedDescription(patternProperty.get().nid())
+                                .get().stamp().time();
+
+                        //Check if the timestamp is valid. If it matches Long.MIN_VALUE + 1
+                        if (rawTime == Long.MIN_VALUE + 1) {
+                            return "Premundane";
+                        }
+
+                        LocalDate localDate = Instant.ofEpochSecond(rawTime).atZone(ZoneId.systemDefault()).toLocalDate();
+                        String fqnDateAddedStr = localDate.format(DateTimeFormatter.ofPattern("MMM d, yyyy"));
```

Comment:
So the problem is the hardcoded date format:
"MMM d, yyyy"
This way we'll always get the month first and then the day and then the year. We want something that's not hardcode like that.  Different countries can have it differently, like there are some where dates are shown as the day first, then the month, then the year.

I think we need some way of grabbing the current local and using that.. though I'm not entirely sure how much of a priority is that 
cc @carldea

Reply:
@bhharsh13 I agree with Pedro @dukke .

Can we try the following:
```java
// Java 8 DateTimeFormatter
DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT).withLocale(locale);
DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM).withLocale(locale);
```